students/meet_shah/lesson09/donor_class.py

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env python3
+
+from functools import reduce
+
+class Donor:
+
+    def __init__(self, name, donations=None):
+        self._name = name
+        self._donations = [] if donations is None else [donations]
+
+
+    @property
+    def name(self):
+        return self._name
+
+
+    @property
+    def donations(self):
+        return self._donations
+
+
+    def add_donations(self, donation):
+        self._donations.append(donation)
+
+
+    @property
+    def num_donations(self):
+        return len(self._donations)
+
+
+    @property
+    def total_donations(self):
+        #return reduce(lambda a, x: a+x, self._donations, 0)
+        return sum(self._donations)
```

Comment:
kinda makes you see why Guido said: 

"most uses of reduce are just sum() anyway"

:-)